npm/vite-dev-server/src/makeCypressPlugin.ts

```diff
@@ -62,11 +68,45 @@ export const makeCypressPlugin = (
 
       server.middlewares.use(`${base}index.html`, (req, res) => res.end(transformedIndexHtml))
     },
-    handleHotUpdate: () => {
-      // restart tests when code is updated
-      devServerEvents.emit('dev-server:compile:success')
+    handleHotUpdate: ({ server, file }) => {
+      debug('handleHotUpdate - file', file)
+      // get the graph node for the file that just got updated
+      let moduleImporters = server.moduleGraph.fileToModulesMap.get(file)
+      let stopIteration = false
+
+      // until we reached a point where the current module is imported by no other
+      while (moduleImporters && moduleImporters.size) {
```

Comment:
This should probably have a max number of steps. We don't need to go through a dependency tree of 1 million for example